test/realizations/catchments/Partition_Parser_Test.cpp

```diff
@@ -0,0 +1,67 @@
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include "Partitions_Parser.hpp"
+
+using namespace realization;
+
+class PartitionsParserTest: public ::testing::Test {
+
+    protected:
+
+    PartitionsParserTest() {
+
+    }
+
+    ~PartitionsParserTest() override {
+
+    }
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupArbitraryExampleCase();
+
+};
+
+void PartitionsParserTest::SetUp() {
+    setupArbitraryExampleCase();
+}
+
+void PartitionsParserTest::TearDown() {
+
+}
+
+void PartitionsParserTest::setupArbitraryExampleCase() {
+
+}
+
+TEST_F(PartitionsParserTest, TestFileReader)
+{
+  const std::string file_path = "/home/shengting.cui/MPI/ngen/data/example_partition_config.json";
+  Partitions_Parser partitions_parser = Partitions_Parser(file_path);
+
+  boost::property_tree::ptree loaded_tree;
+  boost::property_tree::json_parser::read_json(file_path, loaded_tree);
+  boost::property_tree::ptree tree = loaded_tree;
+
+  partitions_parser.read_partition_file(tree);
+
+  int i;
+  std::string part_id;
+  for (i = 0; i <= 1; i++)
+  {
+    part_id = std::to_string(i);
+    partitions_parser.get_part_strt(part_id);
+  }
+
+  for (i = 0; i <= 1; i++)
+  {
+    part_id = std::to_string(i);
+    partitions_parser.get_mpi_rank(part_id);
+  }
+
+  
+  ASSERT_TRUE(true);
```

Comment:
I think these tests are going to need to be reconstructed (i.e., related to how the _read_partition_file_ function needs to be renamed) and, more importantly, expanded.  With the importance of distributed processing, and partition parsing being such a key piece of that, we need to have better test coverage.

I am open to handling the expanding the tests separately, going ahead with this after a few small things get fixed, but I think that should be a high priority.

Reply:
I am open for any suggestions.